GVFS/GVFS.Common/Tracing/TelemetryDaemonEventListener.cs

```diff
@@ -96,39 +101,42 @@ private string SerializeMessage(TraceEventMessage message)
 
         private void SendMessage(string message)
         {
-            // Create pipe if this is the first message, or if the last connection broke for any reason
-            if (this.pipeClient == null)
+            lock (this.pipeLock)
             {
-                var pipe = new NamedPipeClientStream(".", this.pipeName, PipeDirection.Out, PipeOptions.Asynchronous);
-
-                // Specify a instantaneous timeout because we don't want to hold up the rest of the
-                // application if the pipe is not available; we will just drop this event.
-                // We let any TimeoutExceptions bubble up and will try again with a new pipe on the next SendMessage call.
-                pipe.Connect(timeout: 0);
+                // Create pipe if this is the first message, or if the last connection broke for any reason
+                if (this.pipeClient == null)
+                {
+                    var pipe = new NamedPipeClientStream(".", this.pipeName, PipeDirection.Out, PipeOptions.Asynchronous);

```

Comment:
Nit: `var` -> `NamedPipeClientStream` (I see the code was like this before too, but might as well clean it up while we're in here)